### ISSUE TITLE
Login Error into BC 16.2 with Runtime App: Description for event […] cannot be found

### ISSUE THREAD

**Issue by Kowa63** [microsoft/AL#5968]
With every update we also provide other partners with a runtime app version of our extension, so they can install and test it together with their own extensions without requiring our licence to do so. Usually that works fine, we have been using this deployment workflow since BC 13.

With BC 16.2, we're experiencing a new issue here.  
-  the normal extension can be published and installed,enables a login and works as designed 
-  the runtime app only allows a login if it is published without installation. If you then try to install it from within the application, the client freezes
- if you publish and install the runtime app, you'll get the generic error message right away when trying to log in.
![grafik](https://user-images.githubusercontent.com/16236672/84743626-80503100-afb2-11ea-8c3b-35e190f5ecf3.png)

This is the event log from the container:
> The description for Event ID 705 from source MicrosoftDynamicsNavServer$BC cannot be found. Either the component that raises this event is not installed on your local computer or the installation is corrupted. You can install or repair the component on the local computer.

> If the event originated on another computer, the display information had to be saved with the event.

> The following information was included with the event: 
Server instance: BC
Category: Runtime
ClientSessionId: 8a4be1ed-e789-4dbd-81eb-433dce2b4772
ClientActivityId: d7970388-e782-4a01-800b-77605073289f
ServerSessionUniqueId: ba265c53-083d-497d-8a4e-5946989d3767
ServerActivityId: bf4ac4ef-ce42-4987-8c8b-82368ceca05c
EventTime: 06/15/2020 08:07:49
Message (UnauthorizedAccessException): This message had personal data removed. The original may still be in transient telemetry. Find it using the transientTelemetryId.
RootException: UnauthorizedAccessException
ExceptionStackTrace:
   at System.IO.MemoryStream.GetBuffer()
   at Microsoft.Dynamics.Nav.XmlMetadata.NavigationDataAggregator.CleanStreamFromInvalidChars(MemoryStream content)
   at Microsoft.Dynamics.Nav.XmlMetadata.NavigationDataAggregator.CreateNavigationData(MemoryStream content)
   at Microsoft.Dynamics.Nav.XmlMetadata.NavigationDataAggregator.TryGetNavigationDataFromDatabase(NavAppRuntimeMetadata appRuntimeMetadata, NavDiagnostics diagnostics, NavigationDefinition& navigationDefinition)
   at Microsoft.Dynamics.Nav.XmlMetadata.NavigationDataAggregator.GetNavigationData(NavAppGroup appGroup, NavDiagnostics diagnostics)
   at Microsoft.Dynamics.Nav.XmlMetadata.NavigationDataAggregator.FillDepartmentWithNavigationData(NavAppGroup appGroup, ActionContainerDefinition departments, NavDiagnostics diagnostics)
   at Microsoft.Dynamics.Nav.XmlMetadata.NavigationDataAggregator.ConstructDepartmentCacheEntry(NavAppGroup appGroup, NavDiagnostics diagnostics)
   at System.Lazy`1.CreateValue()
   at System.Lazy`1.LazyInitValue()
   at Microsoft.Dynamics.Nav.XmlMetadata.NavigationDataAggregator.CachedDepartmentOrConstructNew(NavAppGroup appGroup, NavDiagnostics diagnostics)
   at Microsoft.Dynamics.Nav.XmlMetadata.NavigationDataAggregator.AppendExtensionNavigationData(NavigationDefinition existingDefinitions, NavAppGroup appGroup)
   at Microsoft.Dynamics.Nav.XmlMetadata.MetadataProvider.GetNavigationDepartments(NavAppGroup appGroup)
   at Microsoft.Dynamics.Nav.XmlMetadata.MetadataProvider.GetNavigationFrame(Int32 homepageId, Boolean applyPersonalization)
   at Microsoft.Dynamics.Nav.XmlMetadata.MetadataProvider.GetNavigationFrame(Boolean applyPersonalization, NavProfileKey profileKey)
  at SyncInvokeGetNavigationFrame(Object , Object[] , Object[] )
   at System.ServiceModel.Dispatcher.SyncMethodInvoker.Invoke(Object instance, Object[] inputs, Object[]& outputs)
   at Microsoft.Dynamics.Nav.Service.ServiceOperationInvoker.RunInTransactionCombinator(ServiceOperation innerOperation, NSServiceBase serviceInstance, MethodBase syncMethod, Object[] inputs, Object[]& outputs)
   at Microsoft.Dynamics.Nav.Service.ServiceOperationInvoker.<>c__DisplayClass28_1.<Combine>b__1(NSServiceBase serviceInstance, Object[] inputs, Object[]& outputs)
   at Microsoft.Dynamics.Nav.Service.ServiceOperationInvoker.TransientErrorRetryCombinator(ServiceOperation innerOperation, NSServiceBase serviceInstance, MethodBase syncMethod, Object[] inputs, Object[]& outputs)
   at Microsoft.Dynamics.Nav.Service.ServiceOperationInvoker.<>c__DisplayClass28_1.<Combine>b__1(NSServiceBase serviceInstance, Object[] inputs, Object[]& outputs)
   at Microsoft.Dynamics.Nav.Service.ServiceOperationInvoker.ErrorMappingCombinator(ServiceOperation innerOperation, NSServiceBase serviceInstance, MethodBase syncMethod, Object[] inputs, Object[]& outputs)
CallerStackTrace:
   at Microsoft.Dynamics.Nav.Service.ServiceOperationInvoker.ErrorMappingCombinator(ServiceOperation innerOperation, NSServiceBase serviceInstance, MethodBase syncMethod, Object[] inputs, Object[]& outputs)
   at Microsoft.Dynamics.Nav.Service.ServiceOperationInvoker.<>c__DisplayClass28_1.<Combine>b__1(NSServiceBase serviceInstance, Object[] inputs, Object[]& outputs)
   at Microsoft.Dynamics.Nav.Service.ServiceOperationInvoker.PushPopCombinator(ServiceOperation innerOperation, NSServiceBase serviceInstance, MethodBase syncMethod, Object[] inputs, Object[]& outputs)
   at Microsoft.Dynamics.Nav.Service.ServiceOperationInvoker.<>c__DisplayClass28_1.<Combine>b__1(NSServiceBase serviceInstance, Object[] inputs, Object[]& outputs)
   at Microsoft.Dynamics.Nav.Service.ServiceOperationTracer.TraceScopeCombinator(Category telemetryCategory, ServiceOperation innerOperation, NSServiceBase serviceInstance, MethodBase syncMethod, Object[] inputs, Object[]& outputs)
   at Microsoft.Dynamics.Nav.Service.ServiceOperationInvoker.<>c__DisplayClass28_1.<Combine>b__1(NSServiceBase serviceInstance, Object[] inputs, Object[]& outputs)
   at Microsoft.Dynamics.Nav.Service.ServiceOperationInvoker.<>c__DisplayClass10_0.<PerformanceCounterCombinator>b__0()
   at Microsoft.Dynamics.Nav.Service.ServiceOperationInvoker.PerformanceCounterCombinator(ServiceOperation innerOperation, NSServiceBase serviceInstance, MethodBase syncMethod, Object[] inputs, Object[]& outputs)
   at Microsoft.Dynamics.Nav.Service.ServiceOperationInvoker.<>c__DisplayClass28_1.<Combine>b__1(NSServiceBase serviceInstance, Object[] inputs, Object[]& outputs)
   at Microsoft.Dynamics.Nav.Service.ServiceOperationInvoker.InitClientTelemetryIdsCombinator(ServiceOperation innerOperation, NSServiceBase serviceInstance, MethodBase syncMethod, Object[] inputs, Object[]& outputs)
   at Microsoft.Dynamics.Nav.Service.ServiceOperationInvoker.<>c__DisplayClass28_1.<Combine>b__1(NSServiceBase serviceInstance, Object[] inputs, Object[]& outputs)
   at Microsoft.Dynamics.Nav.Service.ServiceOperationInvoker.TlsClearCombinator(ServiceOperation innerOperation, NSServiceBase serviceInstance, MethodBase syncMethod, Object[] inputs, Object[]& outputs)
   at Microsoft.Dynamics.Nav.Service.ServiceOperationInvoker.<>c__DisplayClass28_1.<Combine>b__1(NSServiceBase serviceInstance, Object[] inputs, Object[]& outputs)
   at Microsoft.Dynamics.Nav.Service.ServiceOperationInvoker.Invoke(Object instance, Object[] inputs, Object[]& outputs)
   at System.ServiceModel.Dispatcher.DispatchOperationRuntime.InvokeBegin(MessageRpc& rpc)
   at System.ServiceModel.Dispatcher.ImmutableDispatchRuntime.ProcessMessage5(MessageRpc& rpc)
   at System.ServiceModel.Dispatcher.MessageRpc.Process(Boolean isOperationContextSet)
   at System.ServiceModel.Dispatcher.MessageRpc.Wrapper.Resume(Boolean& alreadyResumedNoLock)
   at System.ServiceModel.Dispatcher.ThreadBehavior.ResumeProcessing(IResumeMessageRpc resume)
   at Microsoft.Dynamics.Nav.Runtime.NavSynchronizationContext.<>c__DisplayClass1_0.<ClearThreadLocalStorageDelegate>b__0(Object state)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
ProcessId: 332
Tag: 00000HC
ThreadId: 56
CounterInformation: 
CustomParameters: {
}
GatewayCorrelationId: eddb3d6c-f713-4133-b2ca-8c3b3e14eefa

**Comment by RafaelKoch** [comment]
We are one partner affected and need this resolved as quickly as possible, we develop on top of that package and no pipeline or test environment currently works because of this issue.

**Comment by Kowa63** [comment]
Some more details: 
•      Current VSIX provided by the container was used
•	AL runtime set to current 5.2 
•	Current symbols downloaded from container 
•	GetRuntime of current NAVContainerHelper 0.7.0.3 used 

The same runtime app works fine in 16.1 (no breaking changes between these versions), only 16.2 shows this behaviour.
Only after uninstalling the runtime app a login is possible again.

**Comment by rvanbekkum** [comment]
We are also running into a `UnauthorizedAccessException` when opening the client with a (custom) Base Application installed as run-time package; if we install it as a regular package then all works fine.
Installing the package did not raise any problems, but as soon as you try to use it with the client we are presented with the same screen and exception in the event log.

**Comment by Kowa63** [comment]
You have to update to 16.3 for runtime packages to work. For 16.2 no fix was provided for this issue.

**Comment by rvanbekkum** [comment]
Ah, so you mean it is already fixed for 16.3 and up?
The run-time packages we use are indeed currently generated from a 16.2 container. We have to upgrade anyway for other reasons as well, so if that also fixes this issue that would be nice.

**Comment by Kowa63** [comment]
We went through the same workflow as described above when 16.3 arrived, without any problems.

**Comment by RafaelKoch** [comment]
Same here. Problem went away with 16.3.

**Comment by SteveKrisjanovsD365** [comment]
runtime packages also work in 16.4. I was able to install runtime apps exported from 16.2 and install them on 16.4 with no problems.

**Comment by atoader** [comment]
I am closing this issue because it has been resolved. Please open a new issue if you believe this has not been resolved and reference the current issue.
